src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ManifestTest.cs

```diff
@@ -1211,5 +1212,122 @@ public void IntentFilterDataPathsTest ()
+				<property android:name="test-provider-property" android:value="test-provider-property-value" />
+			</provider>
+		""";
+
+		static readonly string expected_service = """
+			<service android:name="TestService">
+				<property android:name="test-service-property" android:value="test-service-property-value" />
+			</service>
+		""";
+
+		static readonly object [] PropertyElementTestSource = [
+			new object [] {typeof (TestActivity), expected_activity },
+			new object [] {typeof (TestApplication), expected_application },
+			new object [] {typeof (TestBroadcastReceiver), expected_receiver },
+			new object [] {typeof (TestContentProvider), expected_provider },
+			new object [] {typeof (TestService), expected_service },
+		];
+
+		[Test]
+		[TestCaseSource (nameof (PropertyElementTestSource))]
+		public void PropertyTest (Type type, string expected)
+		{
+			var cache = new TypeDefinitionCache ();
+			var asm = AssemblyDefinition.ReadAssembly (type.Assembly.Location);
+			var td = asm.MainModule.GetType ($"Xamarin.Android.Build.Tests.ManifestTest/{type.Name}");
+
+			var manifest = new ManifestDocument (null) {
+				PackageName = "dummy.packageid",
+				VersionResolver = new MockVersionResolver (),
+			};
+
+			manifest.Merge (null, cache, [td], null, false, null, null);
+
+			var sb = new StringWriter ();
+			manifest.Save (null, sb);
+
+			var xml = sb.ToString ();
+			StringAssertEx.AreMultiLineContains (expected, xml);
+		}
+
+		[Activity (Name = "TestActivity")]
+		[Property ("test-activity-property", Value = "test-activity-property-value")]
+		public class TestActivity : global::Android.App.Activity
+		{
+		}
+
+		[Application (Name = "TestApplication")]
+		[Property ("test-application-property", Value = "test-application-property-value")]
+		public class TestApplication : global::Android.App.Application
+		{
+		}
+
+		[BroadcastReceiver (Name = "TestBroadcastReceiver")]
+		[Property ("test-broadcast-property", Value = "test-broadcast-property-value")]
+		public class TestBroadcastReceiver : global::Android.Content.BroadcastReceiver
+		{
+		}
+
+		[ContentProvider (["foo"], Name = "TestContentProvider")]
+		[Property ("test-provider-property", Value = "test-provider-property-value")]
+		public class TestContentProvider : global::Android.Content.ContentProvider
+		{
+		}
+
+		[Service (Name = "TestService")]
+		[Property ("test-service-property", Value = "test-service-property-value")]
+		public class TestService : global::Android.App.Service
+		{
+		}
+
+		class MockVersionResolver : IVersionResolver
+		{
+			public int? GetApiLevelFromId (string id) => 99;
+
+			public string GetIdFromApiLevel (string apiLevel) => "API-99";
+		}
 	}
 }
+
+// Dummy types because we don't want to take a dependency on Mono.Android or Java.Interop
+namespace Android.App
+{
+	public class Activity : Java.Interop.IJavaPeerable { }
+	public class Application : Java.Interop.IJavaPeerable { }
+	public class Service : Java.Interop.IJavaPeerable { }
+}
```

Comment:
Should we put these in a separate file, to make them easier to find? I could see another test wanting to use these at some point, and they are useable within the entire project.